### PR TITLE
[MNO-833] Security Remediation

### PR DESCRIPTION
**Pull request by ouranos**
* Remove deprecated gulp-utils 
* Update a bunch of npm dependencies
* Replace sprity with spritesmith and regenerate sprites
* Fix karma specs

### CODE REVIEW

File: conf/gulp.conf.js [hunk 1]
```diff
@@ -50,7 +51,7 @@ for (const pathName in exports.paths) {
 exports.exitOnError = false;
 exports.errorHandler = function (title) {
   return function (err) {
-    gutil.log(gutil.colors.red(`[${title}]`), err.toString());
+    log(colors.red(`[${title}]`), err.toString());
```
Comment (alexmaestrano) [inline conf/gulp.conf.js]: Just making sure that brackets have to be in a string here?

File: src/images/sprites/mnoe-sprites.less [hunk 2]
```diff
@@ -1,45 +1,66 @@
...
+To change `display` (e.g. `display: inline-block;`), we suggest using a common CSS class:
+
+// CSS
+.icon {
+  display: inline-block;
 }
 
+// HTML
+<i class="icon icon-home"></i>
+*/
```
Comment (alexmaestrano) [inline src/images/sprites/mnoe-sprites.less]: 👍

File: src/images/sprites/mnoe-sprites.less [hunk 3]
```diff
@@ -1,45 +1,66 @@
...
 .mnoe-icon-cc-icons-mno-cc-mastercard {
-  background-position: -0px -60px;
+  background-image: url(../images/sprites/mnoe-sprites.png);
+  background-position: -140px -78px;
   width: 46px;
   height: 30px;
 }
 .mnoe-icon-cc-icons-mno-cc-visa {
-  background-position: -0px -90px;
+  background-image: url(../images/sprites/mnoe-sprites.png);
```
Comment (alexmaestrano) [inline src/images/sprites/mnoe-sprites.less]: Asking for an Angular learning opportunity. Can you also access assets through `@/images/sprites/mnoe-sprites.png`?

Comment (adamaziz15) [reply src/images/sprites/mnoe-sprites.less]: I thought that was only in vue?

File: conf/gulp.conf.js (hunk 1 above)
Comment (ouranos) [reply conf/gulp.conf.js]: the square brackets? or the back-ticks?

The back-ticks are for interpolation (see [template literals|https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals])
The square brackets were just here for still I think :)

File: src/images/sprites/mnoe-sprites.less (hunk 3 above)
Comment (ouranos) [reply src/images/sprites/mnoe-sprites.less]: It's TS aliases defined with the `paths` property in the `compilerOptions` of the `tsconfig.json` file.
eg:
```json
"compilerOptions": {
  "baseUrl": "./",
      "paths": {
        "@models/*": [
          "src/app/_models/*"
        ],
     ...
  }
}
```

**Pull request merged**